operators/multiclusterobservability/controllers/analytics/analytics_controller.go

```diff
@@ -68,6 +79,63 @@ func (r *AnalyticsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	return ctrl.Result{}, nil
 }
 
+// ensureRightSizingDefaults persists default right-sizing flags when absent and returns the (possibly updated) instance.
+func (r *AnalyticsReconciler) ensureRightSizingDefaults(ctx context.Context, instance *mcov1beta2.MultiClusterObservability, reqLogger logr.Logger) (*mcov1beta2.MultiClusterObservability, error) {
+	// Default-enable analytics right-sizing flags ONLY when absent on fresh installs.
+	// Persist defaults back to the MCO spec so users can later override to true/false explicitly.
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(mcoGVK)
+	key := types.NamespacedName{Name: instance.GetName()}
+
+	if err := r.Client.Get(ctx, key, u); err == nil {
+		// Check if the fields already exist
+		nsEnabled, nsFound, _ := unstructured.NestedBool(u.Object,
+			"spec", "capabilities", "platform", "analytics", "namespaceRightSizingRecommendation", "enabled")
+		virtEnabled, virtFound, _ := unstructured.NestedBool(u.Object,
+			"spec", "capabilities", "platform", "analytics", "virtualizationRightSizingRecommendation", "enabled")
+
+		// Only patch if at least one field is missing
+		if !nsFound || !virtFound {
+			// Build a minimal patch that only contains the analytics fields we want to set
+			patchData := map[string]interface{}{
+				"spec": map[string]interface{}{
+					"capabilities": map[string]interface{}{
+						"platform": map[string]interface{}{
+							"analytics": map[string]interface{}{},
+						},
+					},
+				},
+			}
+
+			analytics := patchData["spec"].(map[string]interface{})["capabilities"].(map[string]interface{})["platform"].(map[string]interface{})["analytics"].(map[string]interface{})
```

Comment:
This is very difficult to read for me. These are type assertions/conversions right? You know that it can panic right?

Reply:
Not working.

Reply:
OK probably because of `RawPatch` (as we discussed in Slack) :thinking: